### PR TITLE
Removed code for GET `validator/blinded_blocks` and also adjusted/removed tests

### LINKED ISSUE #7456: Remove client/server code for deprecated `GET validator/blinded_blocks` endpoint
@ethDreamer Noticed that we still have implementations of this deprecated API endpoint:

https://ethereum.github.io/beacon-APIs/?urls.primaryName=v2.5.0#/Validator/produceBlindedBlock

We can safely delete these now that Electra has shipped on mainnet and nobody is using them.

### PR DESCRIPTION
**Pull request by PoulavBhowmick03**
## Issue Addressed

Fixes #7456 

## Proposed Changes

Remove client/server code for deprecated GET `validator/blinded_blocks` endpoint and subsequently, removed the tests which were just testing the blinded endpoint, and modified other endpoints to use the new GET `/eth/v3/validator/blocks` endpoint

## Additional Info

None

### CODE REVIEW

**Comment by PoulavBhowmick03** [comment]
@michaelsproul @eserilev PTAL at this

File: beacon_node/http_api/tests/tests.rs [hunk 1]
```diff
@@ -4336,16 +4113,19 @@ impl ApiTester {
...
-            .unwrap()
-            .into();
+            .unwrap();
+        Self::check_block_v3_metadata(&metadata, &payload_type);
+
+        let payload: BlindedPayload<E> = match payload_type.data {
+            ProduceBlockV3Response::Blinded(payload) => {
+                payload.body().execution_payload().unwrap().into()
+            }
+            ProduceBlockV3Response::Full(_) => panic!("Expecting a full payload"),
```
Comment (michaelsproul) [inline beacon_node/http_api/tests/tests.rs]:
```suggestion
            ProduceBlockV3Response::Full(_) => panic!("Expecting a blinded payload"),
```

File: beacon_node/http_api/tests/tests.rs [hunk 2]
```diff
@@ -4500,16 +4286,19 @@ impl ApiTester {
...
```
Comment (michaelsproul) [inline beacon_node/http_api/tests/tests.rs]:
```suggestion
            ProduceBlockV3Response::Full(_) => panic!("Expecting a blinded payload"),
```

File: beacon_node/http_api/tests/tests.rs [hunk 3]
```diff
@@ -3469,229 +3469,6 @@ impl ApiTester {
...
-                .unwrap()
-                .into_data();
-            assert_eq!(blinded_block.slot(), slot);
-            self.chain.slot_clock.set_slot(slot.as_u64() + 1);
-        }
-
-        self
-    }
-
-    pub async fn test_blinded_block_production_verify_randao_invalid(self) -> Self {
```
Comment (michaelsproul) [inline beacon_node/http_api/tests/tests.rs:3986]: I need to review this more tomorrow, but maybe we could keep these tests and use blocks_v3 with a non-`None` value to force builder usage

**Review by michaelsproul — CHANGES_REQUESTED** [review]

File: beacon_node/http_api/tests/tests.rs (hunk 3 above)
Comment (PoulavBhowmick03) [reply beacon_node/http_api/tests/tests.rs:3986]: Do I need to keep all of the tests that I removed? Or just this one?

Comment (michaelsproul) [reply beacon_node/http_api/tests/tests.rs:3986]: Almost all of them are useful I think.

**Comment by PoulavBhowmick03** [comment]
@michaelsproul can you take a look at the changes and the implementation now please?

File: beacon_node/http_api/tests/tests.rs [hunk 4]
```diff
@@ -4586,16 +4635,19 @@ impl ApiTester {
...
-            .into();
+            .unwrap();
+        Self::check_block_v3_metadata(&metadata, &payload_type);
+
+        let payload: FullPayload<E> = match payload_type.data {
+            ProduceBlockV3Response::Full(payload) => {
+                payload.block().body().execution_payload().unwrap().into()
+            }
+            ProduceBlockV3Response::Blinded(_) => panic!("Expecting a full payload"),
+        };
```
Comment (eserilev) [inline beacon_node/http_api/tests/tests.rs:4981]: comment for clarity: we want to check that we fallback local EE, so this is equivalent to the previous test format. Some of the other test changes below do something similar. looks good!

File: beacon_node/http_api/tests/tests.rs [hunk 5]
```diff
@@ -4768,16 +4823,19 @@ impl ApiTester {
...
```
Comment (eserilev) [inline beacon_node/http_api/tests/tests.rs:4838]: comment for clarity: we want to check that the the we fallback to the local EE, so this is equivalent to the previous test format. looks good

File: beacon_node/http_api/tests/tests.rs [hunk 6]
```diff
@@ -4857,16 +4915,19 @@ impl ApiTester {
...
```
Comment (eserilev) [inline beacon_node/http_api/tests/tests.rs:4930]: comment for clarity: we want to check that the the we fallback to the local EE, so this is equivalent to the previous test format. looks good

**Review by eserilev — APPROVED** [review]
LGTM!

**Comment by PoulavBhowmick03** [comment]
Any update on this? @michaelsproul 
Would be great, i would love to create a pr for #7478 but this would otherwise cause merge conflicts

File: beacon_node/http_api/tests/tests.rs [hunk 7]
```diff
@@ -3507,31 +3507,58 @@ impl ApiTester {
...
-                .unwrap()
-                .into_data();
+                    let head_block = self.chain.head_beacon_block().clone_as_blinded();
+                    assert_eq!(head_block, signed_blinded_block);
 
-            assert_eq!(head_block.clone_as_blinded(), signed_block);
+                    self.chain.slot_clock.set_slot(slot.as_u64() + 1);
+                }
+                ProduceBlockV3Response::Full(block_contents) => {
+                    assert!(!metadata.execution_payload_blinded);
```
Comment (michaelsproul) [inline beacon_node/http_api/tests/tests.rs:3826]: This test is meant to be checking blinded production, so I think it should error in the `Full` case

File: beacon_node/http_api/tests/tests.rs [hunk 8]
```diff
@@ -3613,18 +3640,27 @@ impl ApiTester {
...
-                .into_data();
-            assert_eq!(blinded_block.slot(), slot);
+                .unwrap();
+
+            match response.data {
+                ProduceBlockV3Response::Blinded(blinded_block) => {
+                    assert_eq!(blinded_block.slot(), slot);
+                }
+                ProduceBlockV3Response::Full(full_block) => {
+                    assert_eq!(full_block.block().slot(), slot);
```
Comment (michaelsproul) [inline beacon_node/http_api/tests/tests.rs:3949]: This should also be unreachable, I think we should panic if we get a `Full` block.

**Review by michaelsproul — CHANGES_REQUESTED** [review]
Looks good on the whole, thanks for sticking with this!

Just a few tweaks and then I think we can merge.

Sorry for the slow review, I've been busy trying to get tree-states across the line and also getting some stuff reading for the next PeerDAS devnet

**Comment by PoulavBhowmick03** [comment]
> Looks good on the whole, thanks for sticking with this!
> 
> Just a few tweaks and then I think we can merge.
> 
> Sorry for the slow review, I've been busy trying to get tree-states across the line and also getting some stuff reading for the next PeerDAS devnet

Ah I see, that's some important stuff to get done with! No issues
Thanks a lot!

File: beacon_node/http_api/tests/tests.rs (hunk 8 above)
Comment (PoulavBhowmick03) [reply beacon_node/http_api/tests/tests.rs:3949]: When I add a panic to this and the other line as well, some pre_merge tests seem to panic as they are receiving full blocks
```sh
failures:

---- tests::blinded_block_production_blinded_payload_premerge stdout ----

thread 'tests::blinded_block_production_blinded_payload_premerge' panicked at beacon_node/http_api/tests/tests.rs:3537:21:
Expecting a blinded block
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::blinded_block_production_full_payload_premerge stdout ----

thread 'tests::blinded_block_production_full_payload_premerge' panicked at beacon_node/http_api/tests/tests.rs:3537:21:
Expecting a blinded block

---- tests::blinded_block_production_no_verify_randao_blinded_payload_premerge stdout ----

thread 'tests::blinded_block_production_no_verify_randao_blinded_payload_premerge' panicked at beacon_node/http_api/tests/tests.rs:3638:21:
Expecting a blinded block

---- tests::blinded_block_production_no_verify_randao_full_payload_premerge stdout ----

thread 'tests::blinded_block_production_no_verify_randao_full_payload_premerge' panicked at beacon_node/http_api/tests/tests.rs:3638:21:
Expecting a blinded block

---- tests::blinded_block_production_with_skip_slots_blinded_payload_premerge stdout ----

thread 'tests::blinded_block_production_with_skip_slots_blinded_payload_premerge' panicked at beacon_node/http_api/tests/tests.rs:3537:21:
Expecting a blinded block

---- tests::blinded_block_production_with_skip_slots_full_payload_premerge stdout ----

thread 'tests::blinded_block_production_with_skip_slots_full_payload_premerge' panicked at beacon_node/http_api/tests/tests.rs:3537:21:
Expecting a blinded block


failures:
    tests::blinded_block_production_blinded_payload_premerge
    tests::blinded_block_production_full_payload_premerge
    tests::blinded_block_production_no_verify_randao_blinded_payload_premerge
    tests::blinded_block_production_no_verify_randao_full_payload_premerge
    tests::blinded_block_production_with_skip_slots_blinded_payload_premerge
    tests::blinded_block_production_with_skip_slots_full_payload_premerge
```
@michaelsproul do take a look

Comment (PoulavBhowmick03) [reply beacon_node/http_api/tests/tests.rs:3949]: i have committed a change where it throws error in case of a Full Block, can you PTAL @michaelsproul @eserilev

**Comment by PoulavBhowmick03** [comment]
@michaelsproul, in order for the tests to pass, what should we do in case of panic for FullBlock?

**Comment by michaelsproul** [comment]
what you suggested about only panicking post Bellatrix sounds good

**Comment by PoulavBhowmick03** [comment]
@michaelsproul can you PTAL now

**Comment by michaelsproul** [comment]
Sorry this got backburnered while we fixed the HTTP tests (harder than expected: https://github.com/sigp/lighthouse/pull/7943). I didn't want to merge it while the tests were disabled.

If you can update it with the latest changes from unstable and make sure the tests are passing, we can review it and merge :pray:

Thanks!

**Comment by PoulavBhowmick03** [comment]
> Sorry this got backburnered while we fixed the HTTP tests (harder than expected: #7943). I didn't want to merge it while the tests were disabled.
> 
> If you can update it with the latest changes from unstable and make sure the tests are passing, we can review it and merge 🙏
> 
> Thanks!

yes i was looking into that PR. sure, i will fetch the changes and work on this!

**Comment by PoulavBhowmick03** [comment]
@michaelsproul i have fixed the merge conflicts, can you take a look if it's fine

**Comment by PoulavBhowmick03** [comment]
Mergify might be getting angry over this PR now 😂
